File: Data Structures & Algorithms/products-of-array-discluding-self/submission-4.java
class Solution {
    public int[] productExceptSelf(int[] nums) {

        /**
            [1,2,4,6]

            [1,1,2,8]

            [48,24,6,1]

            [48,24,12,8]

        **/

        int[] left = new int[nums.length];
        int[] right = new int[nums.length];
        int[] result = new int[nums.length];


        for(int i=0;i<nums.length;i++) {
            int temp = i;
            int value = 1;
            while(temp>0) {
                temp--;
                value = value * nums[temp];
            }
            
            left[i] = value;
        }

        for(int n : left) {
            System.out.println(n);
        }

        for(int i=0;i<nums.length;i++) {
            int temp = i+1;
            int value = 1;
            while(temp<nums.length) {
                value = value * nums[temp];
                temp++;
            }

            right[i] = value;
        }

        for(int n : right) {
            System.out.println(n);
        }

        for(int i=0;i<nums.length;i++) {
            result[i] = left[i] * right[i];
        }

        return result;
    }
}  

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-4.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

from collections import deque
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        

        """
            create queue 
            push the root to queue 

            if length of queue is greater than 0 

            pop from queue. 

            push children to list - 
            push children to queue as well 


            return list 
        """

        queue = deque()
        res = []

        if not root:
            return res

        queue.append(root)
        res.append([root.val])

        while queue:
            inner_rest = []
            
            for i in range(len(queue)):
                
                node = queue.popleft()
                if node.left:
                    inner_rest.append(node.left.val)
                    queue.append(node.left)
                if node.right:
                    inner_rest.append(node.right.val)
                    queue.append(node.right)
            if len(inner_rest) != 0:
                res.append(inner_rest)


        return res



                

                

        
        